netplan/libnetplan.py

```diff
@@ -72,11 +88,22 @@ def _string_realloc_call_no_error(function):
     raise LibNetplanException('Aborting due to string buffer size > 1M')  # pragma: nocover
 
 
+# Specialized exceptions based on err.contents.code or err.contents.message
+GERROR_MESSAGE_FORMAT_EXCEPTION_MAP = {
+    r'.*:\d+:\d+: Error in network definition':
+        LibNetplanSchemaValidationException
+}
+
 def _checked_lib_call(fn, *args):
     err = ctypes.POINTER(_GError)()
     ret = bool(fn(*args, ctypes.byref(err)))
     if not ret:
-        raise LibNetplanException(err.contents.message.decode('utf-8'))
+        exception_cls = LibNetplanException
+        for error_format, error_type in GERROR_MESSAGE_FORMAT_EXCEPTION_MAP:
+            if re.match(error_format, err.contents.message.decode('utf-8')):
```

Comment:
Alternatively, we could raise different `err.contents.code` values to represent different exception types because regex matching of `err.contents.message` format matching could be brittle if message format changes or with internationalization of error messages at some point.